sw/device/tests/spi_host_config_test.c

```diff
@@ -0,0 +1,105 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+#include <assert.h>
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_pinmux.h"
+#include "sw/device/lib/dif/dif_spi_host.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/runtime/print.h"
+#include "sw/device/lib/testing/spi_device_testutils.h"
+#include "sw/device/lib/testing/spi_flash_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+#include "sw/device/lib/testing/test_framework/ottf_utils.h"
+#include "sw/device/tests/spi_host_flash_test_impl.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+static_assert(__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__,
+              "This test assumes the target platform is little endian.");
+
+OTTF_DEFINE_TEST_CONFIG();
+
+enum {
+  kDefaultTimeoutMicros = 50000,
+};
+
+uint8_t backdoor_cpha = UINT8_MAX;
+uint8_t backdoor_cpol = UINT8_MAX;
+uint32_t backdoor_data = UINT32_MAX;
+
+static status_t spi_config_test(dif_spi_host_t *spi);
+
+static status_t configure_pinmux(const dif_pinmux_t *pinmux);
+
+bool test_main(void) {
+  dif_spi_host_t spi_host;
+  mmio_region_t base_addr =
+      mmio_region_from_addr(TOP_EARLGREY_PINMUX_AON_BASE_ADDR);
+
+  dif_pinmux_t pinmux;
+  CHECK_DIF_OK(dif_pinmux_init(base_addr, &pinmux));
+  base_addr = mmio_region_from_addr(TOP_EARLGREY_SPI_HOST1_BASE_ADDR);
+  CHECK_DIF_OK(dif_spi_host_init(base_addr, &spi_host));
+
+  configure_pinmux(&pinmux);
+  for (size_t i = 0; i < 4; ++i) {
+    CHECK_STATUS_OK(spi_config_test(&spi_host));
+  }
+  return true;
+}
+
+static status_t spi_config_test(dif_spi_host_t *spi) {
+  dif_spi_host_config_t config = {
+      .spi_clock = 40000,
+      .peripheral_clock_freq_hz = (uint32_t)kClockFreqUsbHz,
```

Comment:
I first thought this seemed off, but since we're testing SPI_HOST1 which is running at 48 MHz (= USB clock) this is indeed correct. I think it would be worth to add a comment on this.

Reply:
Just an FYI: The correct enum reference in the current implementation is `kClockFreqHiSpeedPeripheralHz`. See https://github.com/lowRISC/opentitan/blob/8f2ad247ced6fde7db1b8ff3dcebe021c7bf8b0a/sw/device/lib/arch/device.h#L96

Reply:
Thanks for catching that, I'll open a PR with a fix.

Reply:
I double-checked, and `kClockFreqHiSpeedPeripheralHz` is not the correct constant, according to this [diagram](https://opentitan.org/documentation/index.html), the SPI_HOST1 shares the same clock domain with the USB.
So, either we rename the constant or leave it as it is.

Reply:
I think the correct thing to do here is:
```
kClockFreqHiSpeedPeripheralHz / 2
```
It happens to be 48 MHz as well but it may not be the same 48 MHz clock. For USB, we do have some clock tracking and tuning logic between USBDEV and AST. I think this doesn't apply to the SPI_HOST1 clock.